tests/components/tasmota/test_switch.py

```diff
@@ -0,0 +1,177 @@
+"""The tests for the MQTT switch platform."""
+import copy
+import json
+
+from hatasmota.const import CONF_ONLINE
+
+from homeassistant.components import switch
+from homeassistant.components.tasmota.const import DEFAULT_PREFIX
+from homeassistant.const import ATTR_ASSUMED_STATE, STATE_OFF, STATE_ON
+
+from .test_common import (
+    help_test_availability,
+    help_test_availability_discovery_update,
+    help_test_availability_when_connection_lost,
+    help_test_discovery_device_remove,
+    help_test_discovery_removal,
+    help_test_discovery_update_unchanged,
+    help_test_entity_id_update_discovery_update,
+    help_test_entity_id_update_subscriptions,
+)
+
+from tests.async_mock import patch
+from tests.common import async_fire_mqtt_message
+from tests.components.switch import common
+
+DEFAULT_CONFIG = {
+    "dn": "My Device",
+    "fn": ["Test", "Beer", "Milk", "Four", "Five"],
+    "hn": "tasmota_49A3BC",
+    "mac": "00000049A3BC",
+    "md": "Sonoff 123",
+    "ofl": "offline",
+    CONF_ONLINE: "online",
+    "state": ["OFF", "ON", "TOGGLE", "HOLD"],
+    "sw": "2.3.3.4",
+    "t": "tasmota_49A3BC",
+    "t_f": "%topic%/%prefix%/",
+    "t_p": ["cmnd", "stat", "tele"],
+    "li": [0, 0, 0, 0, 0, 0, 0, 0],
+    "rl": [1, 0, 0, 0, 0, 0, 0, 0],
+    "se": [],
+    "ver": 1,
+}
+
+
+async def test_controlling_state_via_mqtt(hass, mqtt_mock, setup_tasmota):
+    """Test state update via MQTT."""
+    config = copy.deepcopy(DEFAULT_CONFIG)
+    mac = config["mac"]
+
+    async_fire_mqtt_message(
+        hass,
+        f"{DEFAULT_PREFIX}/{mac}/config",
+        json.dumps(config),
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get("switch.test")
+    assert state.state == "unavailable"
+    assert not state.attributes.get(ATTR_ASSUMED_STATE)
+
+    async_fire_mqtt_message(hass, "tasmota_49A3BC/tele/LWT", "online")
+    state = hass.states.get("switch.test")
+    assert state.state == STATE_OFF
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6285789)